### PR TITLE
fix(Switch Node): Maintain output connections

### LINKED ISSUE #11587: When removing a switch rule the outputs changes
### Bug Description

Hi!

I deleted a switch rule and all the outputs are changed.

### To Reproduce

When you create a switch with say for example 3 outputs, pointing to different nodes say output 1 pointing to node 1 etc etc. When I remove for example the first rule leaving only rule 2 and 3 the output changes so output 1 is removed but output 2 is pointing then to node 1 and output 3 is pointing to node 2 and there is a new output created.

### Expected behavior

I would expect when I remove rule 1 the output 1 is removed and output 2 and 3 stay connected to node 2 and 3.

### Operating System

docker compose

### n8n Version

1.65.2

### Node.js Version

?

### Database

SQLite (default)

### Execution mode

main (default)

### PR DESCRIPTION
**Pull request by michael-radency**
## Summary

maintain connection on re-order
remove connection if rule for this connection was removed
if fallback set to extra preserve connected node and do not attach it to added rule output
if using expression for setting number of outputs trim connections if their number was reduced

## Related Linear tickets, Github issues, and Community forum posts

https://linear.app/n8n/issue/NODE-1669/switch-node-adding-routing-rule-reconnects-fallback-route

fixes https://github.com/n8n-io/n8n/issues/11587

### CODE REVIEW

File: packages/editor-ui/src/components/NodeSettings.vue [hunk 1]
```diff
@@ -47,11 +43,12 @@ import { useCredentialsStore } from '@/stores/credentials.store';
 import type { EventBus } from 'n8n-design-system';
 import { useExternalHooks } from '@/composables/useExternalHooks';
 import { useNodeHelpers } from '@/composables/useNodeHelpers';
-import { useToast } from '@/composables/useToast';
+// import { useToast } from '@/composables/useToast';
```
Comment (ShireenMissi) [inline packages/editor-ui/src/components/NodeSettings.vue:46]: if this is not needed let's just delete it

**Review by ShireenMissi — COMMENTED** [review]
I tested locally and on test instance, the only issue is when you add a new rule and create a new connection then reorder the new output doesn't maintain its connection
on test instance it works once I save the workflow
locally I had to save and refresh the page to get it to work

File: packages/editor-ui/src/stores/workflows.store.ts [hunk 2]
```diff
@@ -996,8 +996,12 @@ export const useWorkflowsStore = defineStore(STORES.WORKFLOWS, () => {
 		workflow.value.nodes = nodes;
 	}
 
-	function setConnections(connections: IConnections): void {
+	function setConnections(connections: IConnections, updateCashedWorkflow = false): void {
 		workflow.value.connections = connections;
+
+		if (updateCashedWorkflow) {
```
Comment (elsmr) [inline packages/editor-ui/src/stores/workflows.store.ts:1002]: cashed -> cached

**Review by elsmr — CHANGES_REQUESTED** [review]
Found a bug when the workflow is not saved.

In general I think it's an OK solution for the Switch node, but it would be great if connections were saved by ID instead of by index to avoid all this complexity. But not clear to me how to do it without a breaking change.

I think you could have similar bugs in other nodes with dynamic outputs such as the Webhook node.

**Review by elsmr — DISMISSED** [review]

File: packages/editor-ui/src/utils/nodeSettingsUtils.ts [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+						reorderedConnections.push(lastConnection);
+					}
+
+					connections[node.name].main = reorderedConnections;
+					return connections;
+				}
+			}
+		}
+	} catch (error) {
+		console.log(error.message);
```
Comment (elsmr) [inline packages/editor-ui/src/utils/nodeSettingsUtils.ts:127]: Could maybe be `captureException` from `@sentry/vue` instead of a log?

**Review by elsmr — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.68.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.68.0)

**Pull request merged**